_posts/2016/2016-02-16-pipeline-elk-test-evaluation-2.md

```diff
@@ -0,0 +1,290 @@
+
+#### Testing
+
+The CI part for our Logstash container is very similiar to that of the Elasticsearch container. The tests we run are different of course, but the process is the same.
+
+### Part 4 - Integrate Docker Containers in Pipeline with Jenkins
+
+#### Logstash
+
+We have several pipeline jobs that run the test suite of the ePages Selenium Framework against all kinds of ePages setups. As a result they produce, amongst a bunch of other files, a log file with the JSON test objects as described in part 1.
+
+In such Jenkins jobs we added a separate build step where we first check that all required environment variables are set. Then we pull the Logstash container from the registry and use the start script to run the container accordingly. Below you can see an extract of the console output in verbose mode.
+
+{% highlight text %}
+=== Start docker container [to-logstash-run-esf-tests-3829] from image [epages/to-logstash:latest] ===
+
+Process logs with pattern:          *esf*.json
+Mount log directory:                /home/jenkins/workspace/Run_ESF_tests/esf/esf-epages6-1.15.0-SNAPSHOT/log
+Mount config directory:             /home/jenkins/workspace/Run_ESF_tests/to-logstash/config
+Set logstash input types:           log,esf
+Set logstash output types:          log,elasticsearch,verbose
+Use logstash env file:              env-esf.list
+Use logstash conf file:             logstash-esf.conf
+Use info log file:                  logstash-info.json
+Use error log file:                 logstash-error.json
+Use elasticsearch template file:    template-esf.json
+Set elasticsearch hosts:            [ 'cd-vm-docker-host-001.intern.epages.de:9200' ]
+Set elasticsearch index:            esf-cdp-ui-tests
+Set elasticsearch document type:    6.17.40
+
+--- Start configuration is applied.
+
+a8fa29d74ef97832fcfbc3a0722b728a465244263c9d482b6eec6357d184555b
+
+--- Logstash finished processing and container is terminated.
+
+=== No need to stop not running docker container [to-logstash-run-esf-tests-3829] ===
+
+=== Remove existing docker container [to-logstash-run-esf-tests-3829] ===
+
+to-logstash-run-esf-tests-3829
+{% endhighlight %}
+
+All shipped test objects are saved to a Logstash info log, which is archived as a build artifact in the post-build section of Jenkins.
+
+#### Elasticsearch
+
+For our Elasticsearch Docker cluster we configured a new Jenkins job, which ensured that always the latest stable version of our image is used. We made sure to mount several host directories so that the Elasticsearch data, config and logs are stored on the VM. By firing up multiple Elasticsearch node containers joining the same cluster we achieved load-balance and shard redundancy.
+
+### Part 5 - Use the Elasticsearch Client to Evaluate the Test Results
```

Comment:
Use the Elasticsearch Client to Evaluate the Test Results > Use the Elasticsearch client to evaluate the test results